test/batch_node.jl

```diff
@@ -0,0 +1,129 @@
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_MAIN_NODE_INDEX DNE" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id"
+        ) do
+            @test_throws ErrorException AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES DNE" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+        ) do
+            @test_throws KeyError AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NODE_INDEX != AWS_BATCH_JOB_MAIN_NODE_INDEX" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_NODE_INDEX" => 0,
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1
+        ) do
+            @test_throws ErrorException AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES -- String" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NUM_NODES" => "foobar"
+        ) do
+            @test_throws ArgumentError AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES -- Int" begin
+        expected_workers = 9
+
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NUM_NODES" => expected_workers + 1  # Add one to account for the manager
+        ) do
+            result = AWSBatchNodeManager()
+
+            @test result.num_workers == expected_workers
+            @test result.timeout == AWSClusterManagers.AWS_BATCH_NODE_TIMEOUT
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES -- Int" begin
+        expected_workers = 9
+        expected_timeout = Second(5)
+
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NUM_NODES" => expected_workers + 1  # Add one to account for the manager
+        ) do
+            result = AWSBatchNodeManager(; timeout=expected_timeout)
+
+            @test result.num_workers == expected_workers
+            @test result.timeout == expected_timeout
+        end
+    end
+end
+
+@testset "parse_job_id" begin
+    passing_cases = [
+        string(repeat("a", 36), "#123"),
+        string("0-", repeat("a", 34), "#123")
+    ]
+
+    @testset "passing case: $(case)" for case in passing_cases
+        str = string("job_id:", case)
+
+        @test AWSClusterManagers.parse_job_id(str) == case
+    end
+
+    failing_cases = [
+        string(repeat("a", 35), "#123"),
+        string(repeat("a", 37), "#123"),
+        string(repeat("!", 36), "#123"),
+        string(repeat("a", 36), "#"),
+        string(repeat("a", 36), "#abc"),
```

Comment:
Would be helpful to add comments as to why these fail or why the other ones pass